src/nvim/event/libuv_process.c

```diff
@@ -101,6 +101,14 @@ static void close_cb(uv_handle_t *handle)
 static void exit_cb(uv_process_t *handle, int64_t status, int term_signal)
 {
   Process *proc = handle->data;
-  proc->status = (int)status;
+#if defined(WIN32)
+  // Use stored/expected signal.
+  term_signal = proc->exit_signal;
+#endif
+  if (term_signal) {
+    proc->status = 128 + term_signal;
+  } else {
+    proc->status = (int)status;
```

Comment:
ternary saves 6 lines without loss of clarity.

Reply:
> Why is it named both `term_signal` and `exit_signal`? Should be the same in all places.

`term_signal` is what we're getting in the libuv callback.
I've used `exit_signal` mostly to make it distinct - i.e. it's what we're setting, not necessarily the final signal.  Do you still think it should be the same name?